web/src/drafts.ts

```diff
@@ -640,7 +639,7 @@ export function format_draft(draft: LocalStorageDraftWithId): FormattedDraft | u
 
         let draft_topic_display_name = draft.topic;
         let is_empty_string_topic = false;
-        if (draft.topic === "" && !realm.realm_mandatory_topics) {
+        if (draft.topic === "" && stream_data.can_use_empty_topic(draft.stream_id)) {
```

Comment:
This is a specific code path where it's definitely possible for `stream_id` to be null -- basically if you saved a draft while that input was unset.